app/controllers/products_controller.rb

```diff
@@ -11,12 +11,31 @@ class ProductsController < ApplicationController
       The prices are defined in the 'pricings' list, each 'pricing' holds the
       price and quantity available at that price
     EON
-    param :path, :id, :integer, :required, 'The product ID'
+    param :path, :id, :integer, :required, 'Product ID'
+    response :ok, 'Success', :readProduct
     response :not_found
   end
 
   swagger_api :index do
     summary 'Show all products'
+    response :ok, 'Success', :readProduct
+  end
+
+  swagger_model :readProduct do
+    description 'A Product object'
+    property :id, :integer, :optional, 'Product ID'
+    property :name, :string, :optional, 'Product name'
+    property :quantity, :integer, :optional, 'Total quantity'
+    property :available_quantity, :integer, :optional, 'Available quantity'
+    property :pricings, :array, :optional, 'Pricings', {'items' => {'$ref' => 'readPricing'}}
```

Comment:
Line is too long. [93/80]<br>Redundant curly braces around a hash parameter.<br>Space inside { missing.<br>Space inside } missing.